worlds/sc2/options.py

```diff
@@ -1093,6 +1092,16 @@ def get_excluded_missions(world: 'SC2World') -> Set[SC2Mission]:
     # Omitting missions not in enabled campaigns
     for campaign in disabled_campaigns:
         excluded_missions = excluded_missions.union(campaign_mission_table[campaign])
+    # Omitting unwanted mission variants
+    if get_option_value(world, "enable_race_swap") == EnableRaceSwapVariants.option_pick_one:
+        swaps: List[SC2Mission] = list([mission for mission in set(SC2Mission).difference(excluded_missions).intersection(get_missions_with_any_flags_in_list(MissionFlag.HasRaceSwap|MissionFlag.RaceSwap))])
```

Comment:
This is still random independent of seed because the order of iterating over a set is non-deterministic. I recommend keeping things in lists and using list comprehension, e.g.
```python
swaps = [
    mission for mission in SC2Mission
    if mission not in excluded_missions
    and mission.flags & (MissionFlag.HasRaceSwap|MissionFlag.RaceSwap)
]
```

Reply:
Aha! That's what I was trying to do for like half an hour but something in my syntax was screwing up so only half the logic got applied. Thanks!